File: collatz/collatz_recursion.py
from time import perf_counter

def collatz(number, steps=0):
    if number == 1:
        return steps
    elif number % 2 == 0:
        steps += 1
        return collatz(number/2, steps)
    else:
        steps += 1
        return collatz(number * 3 + 1, steps)
        
num_to_check = int(input("Please enter a number to check: "))
time_start = perf_counter()
print(f"The number of steps for {num_to_check} is {collatz(num_to_check)}")
time_end = perf_counter()

print(f"The time elapsed in seconds: {time_end-time_start}")

File: collatz/collatz_recursion.c
/*
Andrew Mark Dale
Collatz Conjecture using recursion
*/

#include <stdio.h>
#include <time.h>

int collatz(long long num, int steps){
    if(num == 1){
        return steps;
    }
    else if(num % 2 == 0){
        steps++;
        collatz(num / 2, steps);
    }
    else{
        steps++;
        collatz(3*num + 1, steps);
    }
}

int main(){
    long long inputNum;
    printf("Enter a number: ");
    scanf("%lld", &inputNum);

    clock_t tic = clock();
    printf("The number of steps for %lld is %d\n", inputNum, collatz(inputNum, 0));
    clock_t toc = clock();
    printf("Elapsed: %f seconds\n", (double)(toc - tic) / CLOCKS_PER_SEC);
    return 0;
}